File: Android/app/src/main/java/alexa4/friendphoto/controllers/FriendsController.java
package alexa4.friendphoto.controllers;

import android.util.Log;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;

import alexa4.friendphoto.models.Friend;


/**
 * Controller that contains list of friends and can restore it from json string
 */
public class FriendsController {
    private static final String TAG = "FriendsController";
    public final ArrayList<Friend> mFriends = new ArrayList<>();


    /**
     * Set up friends list from input json string
     *
     * @param jsonString input string with friends
     */
    public void setFriends(String jsonString) {
        Log.d(TAG, jsonString);
        mFriends.clear();
        try {
            JSONObject json = new JSONObject(jsonString).getJSONObject("response");
            int count = json.getInt("count");
            JSONArray array = json.getJSONArray("items");
            for (int i = 0; i < count; i++)
                mFriends.add(new Friend(array.getJSONObject(i)));

        } catch (JSONException e) {
            e.printStackTrace();
            mFriends.clear();
        }
    }

    public void dispose() {
        mFriends.clear();
    }
}

File: Flutter/friend_photo_flutter/lib/src/controllers/friends_controller.dart
import 'dart:convert';

import 'package:friend_photo_flutter/src/models/friend.dart';

/// Controller that contains list of friends and can restore it from json string
class FriendsController {
  static final JsonCodec _codec = JsonCodec();
  final List<Friend> friends = [];

  /// Check is [friends] already downloaded
  bool isLoaded() => friends.length != 0;

  /// Set up friends list from input [jsonString]
  void setFriends(String jsonString) {
    print(jsonString);
    friends.clear();

    try {
      Map<String, dynamic> json = _codec.decode(jsonString)['response'];
      List list = json['items'];
      list.forEach((friend) => friends.add(new Friend.fromJson(friend)));
    } catch (e) {
      print(e);
      friends.clear();
    }
  }

  /// Set up friends list from DB
  void setFromDb(List<Friend> newFriends) {
    friends.addAll(newFriends);
  }

  /// Free memory
  void dispose() {
    friends.clear();
  }
}
